src/pudl/helpers.py

```diff
@@ -1527,6 +1528,19 @@ def dedupe_n_flatten_list_of_lists(mega_list):
     return list({item for sublist in mega_list for item in sublist})
 
 
+def flatten(xs: list) -> Generator:
+    """Flatten an irregular (arbitrarily nested) list of lists.
+
+    Inspiration from: https://stackoverflow.com/questions/2158395/flatten-an-irregular-
+    arbitrarily-nested-list-of-lists
```

Comment:
i though my reference would be helpful... but how do I make the pre-commit hooks to stop breaking this link into two lines?

Reply:
yes to the former. the pre-commit hooks pull it back up to the first line and give it a line break. I can try the latter.

Reply:
omigosh i bashed my head against this link formatting but really the pre-commit hooks didn't like the from: